File: src/API/src/Modules/Clients/Clients.Application/DTOs/CreateClientProductRequest.cs
namespace Clients.Application.DTOs
{
    public class CreateClientProductRequest
    {
        public Guid ClientId { get; set; }
        public Guid ProductId { get; set; }
        public string RelationshipStatus { get; set; } = string.Empty;
        public Guid? OpportunityId { get; set; }
        public Guid? OwnerUserId { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Notes { get; set; }
    }
}

File: src/ClientApp/src/app/pages/clients/dtos/create-client-product.request.ts
export interface CreateClientProductRequest {
    clientId: string;
    productId: string;
    relationshipStatus: string;
    opportunityId?: string | null;
    ownerUserId?: string | null;
    startDate?: string | null;
    endDate?: string | null;
    notes?: string | null;
}
